Market Tools/macd_strategy.py

```diff
@@ -0,0 +1,147 @@
+# Import the required libraries
+import numpy as np
+import pandas as pd
+import hvplot.pandas
+from pathlib import Path
+from datetime import date
+from pandas.core.frame import DataFrame
+import yfinance as yf
+import os
+import pricing
+
+def analyze_macd(signals_df:DataFrame):
+  # Calculate the MACD and Signal Line Indicators
+  # Calculate the short term exponential moving average (EMA)
+  signals_df["short_ema"] = signals_df.Close.ewm(span=12, adjust=False).mean()
```

Comment:
[Low priority, not for this PR] : In figure coding consider using try/except/finally blocks to _error handle_ your code. This will enable you to handle unexpected situations without the app crashing and burning.  I haven't done this in my own code as an example to show. But it's an important thing to learn that will serve you well.